wiki/content/graphql/lambda/overview.md

```diff
@@ -0,0 +1,195 @@
+
+1. You can add lambda fields to your types and interfaces
+
+```graphql
+type MyType {
+    ...
+    customField: String @lambda
+}
+```
+
+2. You can add lambda queries to the Query type
+
+```graphql
+type Query {
+    myCustomQuery(...): QueryResultType @lambda
+}
+```
+
+3. You can add lambda mutations to the Mutation type
+
+```graphql
+type Mutation {
+    myCustomMutation(...): MutationResult @lambda
+}
+```
+
+## Defining lambda resolvers in JavaScript
+
+A lambda resolver is a user-defined JavaScript function that performs custom actions over the GraphQL types, interfaces, queries, and mutations. There are two methods to register JavaScript resolvers:
+
+- `self.addGraphQLResolvers`
+- `self.addMultiParentGraphQLResolvers`
+
+### addGraphQLResolvers
+
+The `self.addGraphQLResolvers` method takes an object as an argument, which maps a resolver name to the resolver function that implements it. The resolver functions registered using `self.addGraphQLResolvers` receive `{ parent, args, graphql, dql }` as argument:
+
+- `parent`, the parent object for which to resolve the current lambda field registered using `addGraphQLResolver`.
+Available only for types and interfaces (`null` for queries and mutations)
+- `args`,  the set of arguments for lambda queries and mutations (`null` for types and interfaces)
+- `graphql`, a function to execute auto-generated GraphQL API calls from the lambda server
+- `dql`, provides an API to execute DQL from the lambda server
+
+{{% notice "tip" %}}
+`self.addGraphQLResolvers` is the default choice for registering resolvers when not all the parent objects of a lambda field are required during resolution.
+{{% /notice %}}
+
+Each resolver function should return data in the exact format as the return type of GraphQL field, query, or mutation for which it is being registered.
+
+In the following example, the resolver function `myTypeResolver` registered for the `customField` field in `MyType` returns a string because the return type of that field in the GraphQL schema is `String`:
+
+```javascript
+const myTypeResolver = ({parent: {customField}}) => `My value is ${customField}.`
+
+self.addGraphQLResolvers({
+    "MyType.customField": myTypeResolver
+})
+```
+
+Another resolver example using a `graphql` call:
+
+```javascript
+async function todoTitles({ graphql }) {
+  const results = await graphql('{ queryTodo { title } }')
+  return results.data.queryTodo.map(t => t.title)
+}
+
+self.addGraphQLResolvers({
+  "Query.todoTitles": todoTitles
+})
+```
+
+### addMultiParentGraphQLResolvers
+
+The `self.addMultiParentGraphQLResolvers` is useful in scenarios where you want to perform computations involving all the parent objects returned from Dgraph for a lambda field.
+This is only useful for lambda fields and not for queries or mutations. 
+
+{{% notice "tip" %}}
+This method provides a much better performance if you are able to combine multiple requests together.
```

Comment:
technically, this is not true, as we club the queries even with the other function. I think the tip I mentioned before is a more accurate description.

Reply:
Tip removed